### PR TITLE
Revise content to match new troubleshooting guidelines

### PR DESCRIPTION
**Pull request by marciw**
[Preview](https://elasticsearch_bk_118033.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/task-queue-backlog.html)

Revise to match new troubleshooting guidelines, mostly to create an example to link to from the guidelines.

- Restructure and add new elements to reflect the new symptom ref content type
- Edit generally for conciseness and clarity

Still under discussion: "symptom" sections

### CODE REVIEW

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 1]
```diff
@@ -1,103 +1,117 @@
...
+A backlogged task queue can prevent tasks from completing and lead to an 
+unhealthy cluster state. Contributing factors include resource constraints, 
+a large number of tasks triggered at once, and long-running tasks.
 
 [discrete]
 [[diagnose-task-queue-backlog]]
-==== Diagnose a task queue backlog
+==== Diagnose a backlogged task queue
 
 **Check the thread pool status**
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:18]: I think ideally we'd have an introductory sentence before this and reformat these as child headings

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 2]
```diff
@@ -1,103 +1,117 @@
 [[task-queue-backlog]]
 === Task queue backlog
 
-A backlogged task queue can prevent tasks from completing and put the cluster
-into an unhealthy state. Resource constraints, a large number of tasks being
-triggered at once, and long running tasks can all contribute to a backlogged
-task queue.
+*******************************
+*Product:* Elasticsearch +
+*Deployment type:* Elastic Cloud (hosted or self-managed), self-managed +
```
[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:6]: given that this tutorial uses cat thread pool apis it's def not for serverless ... but probably for everything else

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 3]
```diff
@@ -1,103 +1,117 @@
...
 
 A <<high-cpu-usage,depleted thread pool>> can result in
 <<rejected-requests,rejected requests>>. 
 
-Thread pool depletion might be restricted to a specific <<data-tiers,data tier>>. If <<hotspotting,hot spotting>> is occuring, one node might experience depletion faster than other nodes, leading to performance issues and a growing task backlog.
-
-You can use the <<cat-thread-pool,cat thread pool API>> to see the number of
-active threads in each thread pool and how many tasks are queued, how many
-have been rejected, and how many have completed.
+Use the <<cat-thread-pool,cat thread pool API> to monitor
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:23]: borky link
```suggestion
Use the <<cat-thread-pool,cat thread pool API>> to monitor
```

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 4]
```diff
@@ -1,103 +1,117 @@
...
 [source,console]
 ----
 GET /_cat/thread_pool?v&s=t,n&h=type,name,node_name,active,queue,rejected,completed
 ----
 
-The `active` and `queue` statistics are instantaneous while the `rejected` and
-`completed` statistics are cumulative from node startup.
+* Look for high `active` and `queue` metrics, which indicate potential bottlenecks.
+* Analyze whether thread pool issues are specific to a <<data-tiers,data tier>> or
+caused by uneven node resource utilization such as <<hotspotting,hot spotting>>.
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:33]: uneven node resource utilization is synonymous with hot spotting. would rephrase this to not be "caused by uneven node resource utilization" but instead as something to look for. hack edit:

```suggestion
* Analyze whether thread pool issues are specific to a <<data-tiers,data tier>>.
* Analyze whether a specific node's thread pool is depleting faster than others. This can be caused by uneven node resource utilization, also known as <<hotspotting,hot spotting>>.
```

stray thoughts: 
- wonder if we should even introduce hot spotting here or leave it to the resolution
- wonder if "analyze" is too loaded of a word (overthinking here). people might think it's harder than it is, when it's really a "check if"
- little red flag in the back of my mind here - in some cases, we can already tie diagnostics to resolutions at this stage. should we be tying them together somehow? 
- is hotspotting missing from the recommendations?

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 5]
```diff
@@ -1,103 +1,117 @@
...
 ----
 
-Check the `timeInQueue` to identify tasks that are taking an excessive amount 
-of time to complete.
+Tasks with a high `timeInQueue` value are likely contributing to the backlog.
 
 [discrete]
 [[resolve-task-queue-backlog]]
-==== Resolve a task queue backlog
+==== Recommendations
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:88]: Re: the recommendations quandary

I think that we shouldn't sacrifice clarity for following linguistic rules, so a noun is ok here.

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 6]
```diff
@@ -1,103 +1,117 @@
...
+Long-running tasks can also cause a backlog. Use the <<tasks,task
+management API>> to check for excessive `running_time_in_nanos` values:
 
 [source,console]
 ----
 GET /_tasks?pretty=true&human=true&detailed=true
 ----
 
-If a particular `action` is suspected, you can filter the tasks further. The most common long-running tasks are <<docs-bulk,bulk index>>- or search-related.
+You can filter on a specific `action`, such as <<docs-bulk,bulk indexing>> or search-related tasks.
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:58]: I think it's nice to explain why we suggest filtering on these (they tend to be long-running)

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 7]
```diff
@@ -1,103 +1,117 @@
...
+----
+
+For more information, see <<settings>>.
 
 **Cancel stuck tasks**
 
-If you find the active task's hot thread isn't progressing and there's a backlog, 
-consider canceling the task. 
\ No newline at end of file
+If an active task's <<diagnose-hot-thread,hot thread>> shows no progress, consider canceling the task.
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:107]: I wonder how we can connect a hot thread to a task. is that done with the cat thread pool api?

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 8]
```diff
@@ -1,103 +1,117 @@
...
+
+[discrete]
+==== Resources
+
+Related symptoms:
+
+* <<high-cpu-usage,High CPU usage>>
+* <<rejected-requests,Rejected requests>>
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:115]: add hot spotting?

**Review by shainaraskas — COMMENTED** [review]
thanks for taking on this first edit!! 

some thinkies for you

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 9]
```diff
@@ -1,103 +1,117 @@
 [[task-queue-backlog]]
 === Task queue backlog
```
[the remark this answers is not in the document]
Comment (marciw) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:2]: Agreed... except I did it this way initially and then thought "hmm, maybe for SEO I should cover both bases." Still, I agree and am glad for the second!

[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:2]: oo smart also not a bad plan - ok with either

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 10]
```diff
@@ -1,103 +1,117 @@
...
-In some cases, increasing the thread pool size might help.
-For example, the `force_merge` thread pool defaults to a single thread.
-Increasing the size to 2 might help reduce a backlog of force merge requests.
+For example, the `force_merge` thread pool defaults to a single thread. 
+Increasing the size to 2 in `elasticsearch.yml` might help reduce a backlog 
+of force merge requests:
+
+[source,yaml]
+----
+thread_pool.force_merge.size: 2
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:100]: unfortunately we need to figure out whether ESS supports this setting. suspect the answer is no

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (marciw) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:6]: I wonder whether there's a case for just saying "All types except serverless" -- seems like that might be the case often enough?

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc (hunk 10 above)
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:100]: image
correct

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc (hunk 7 above)
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:107]: do we know how to cancel tasks

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:6]: :2c: until we clean up our usage of elastic cloud it's better to be specific (to help later contributors)

[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:6]: (or do me a favor and add it as a comment)

**Comment by marciw** [comment]
@shainaraskas TY for the thorough review! I think I've addressed (or skirted) all of your suggestions... at least to the point of "good enough for the purposes of an example"? 🤔

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (shainaraskas) [reply docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc]: yeah I think we need to see how some conversations around that official messaging goes - but agree we should be able to link to a source of truth. afraid of putting this list together now because it is actively being discussed

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 11]
```diff
@@ -45,7 +51,15 @@ progression and ensure it has sufficient resources:
 GET /_nodes/hot_threads
 ----
 
-**Identify long-running node tasks**
+Although the hot threads API response does not list the specific tasks running on a thread, 
+it provides a summary of the thread's activities. You can correlate a hot threads response 
+with a <<tasks,task management API response>> to identify any overlap with specific tasks. For 
+example, if the hot threads response indicates the thread is `performing a search query`, you can 
+<<diagnose-task-queue-long-running-node-tasks,check for long-running search tasks>> via the task management API.
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:58]: avoid latinisms

```suggestion
<<diagnose-task-queue-long-running-node-tasks,check for long-running search tasks>> using the task management API.
```

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 12]
```diff
@@ -81,37 +100,45 @@ in cluster state synchronization:
...
 
-<<reduce-cpu-usage>> or increase thread pool sizes. 
+[discrete]
+[[resolve-task-queue-backlog-resources]]
+===== Increase available resources
 
-For example, the `force_merge` thread pool defaults to a single thread. 
-Increasing the size to 2 in `elasticsearch.yml` might help reduce a backlog 
-of force merge requests:
+If tasks are progressing slowly, try reducing <<reduce-cpu-usage,CPU usage>>.
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:116]:
```suggestion
If tasks are progressing slowly, try <<reduce-cpu-usage,reducing CPU usage>>.
```

File: docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc [hunk 13]
```diff
@@ -81,37 +100,45 @@ in cluster state synchronization:
...
+
+If an active task's <<diagnose-task-queue-hot-thread,hot thread>> shows no progress, consider <<task-cancellation,canceling the task>>.
 
-**Cancel stuck tasks**
+[discrete]
+[[resolve-task-queue-backlog-hotspotting]]
+===== Address hot spotting
 
-If an active task's <<diagnose-hot-thread,hot thread>> shows no progress, consider canceling the task.
+If a specific node's thread pool is problematic, try addressing
```
Comment (shainaraskas) [inline docs/reference/troubleshooting/common-issues/task-queue-backlog.asciidoc:131]: `problematic` is a little loaded
```suggestion
If a specific node's thread pool is depleting faster than others, try addressing
```

**Review by shainaraskas — APPROVED** [review]
beautiful and delightful to behold! some baby cosmetic thingies, then I think we're ready to link to it in the guidance
